### PR TITLE
Jcs/async query tests

### PR DESCRIPTION
**Pull request by jSylvestre**

### CODE REVIEW

**Comment by jSylvestre** [comment]
@srkirkland I tried to base these tests on the code here:
https://msdn.microsoft.com/en-us/library/dn314429.aspx

But, when I run the async one, I get the exception:
Result Message:	The source IQueryable doesn't implement IAsyncEnumerable<Payments.Core.Models.Invoice>. Only sources that implement IAsyncEnumerable can be used for Entity Framework asynchronous operations.

The only examples I can seem to find use IDbAsync...

File: tests/Payments.Tests/project.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "xunit.runner.visualstudio": "2.1.0",
     "Payments": "1.0.0-*",
     "Moq": "4.5.28",
-    "Shouldly": "2.8.2"
+    "Shouldly": "2.8.2",
+    "EntityFramework": "6.1.3"
```
Comment (jSylvestre) [inline tests/Payments.Tests/project.json]: @srkirkland Also had to grab this, which probably isn't right.

Comment (srkirkland) [inline tests/Payments.Tests/project.json]: @jSylvestre definitely don't do this, we don't want that version of EF

File: tests/Payments.Tests/Helpers/AsyncHelper.cs [hunk 2]
```diff
@@ -41,13 +42,18 @@ public Task<object> ExecuteAsync(Expression expression, CancellationToken cancel
             return Task.FromResult(Execute(expression));
         }
 
+        public IAsyncEnumerable<TResult> ExecuteAsync<TResult>(Expression expression)
+        {
+            throw new System.NotImplementedException();
```
Comment (jSylvestre) [inline tests/Payments.Tests/Helpers/AsyncHelper.cs]: @srkirkland Do you know what this should return?

**Pull request closed without merging**